test/parallel_api/iterator/iterators.pass.cpp

```diff
@@ -279,6 +279,45 @@ struct test_transform_iterator {
     }
 };
 
+struct test_permutation_iterator
+{
+    template <typename T1, typename T2>
+    void
+    operator()(::std::vector<T1>& in1, ::std::vector<T2>& in2)
+    {
+        T1 iota_max = ::std::numeric_limits<T1>::max() < in1.size() ? ::std::numeric_limits<T1>::max() : in1.size();
```

Comment:
This is to handle the case where the type `T1` is too short to hold the size of `in1`.  In one test case  `int16_t` is used when n = 100000.  That results in negative values in the result of the algorithms called due to overflow.  The value of `iota_max` will never be greater than the size of the vector `in1` passed, so there's no risk of a large value causing UB or an out of memory error.

Reply:
In that case of "will never be greater than the size of the vector", probably does it make sense to add an assert?

Reply:
I don't think an assert makes sense.  What the statement is doing is picking the larger of two values, the size of the input vector or the maximum value for the data type.  This is done to avoid overflow in the computation in the case that the max value of T1 is less than the number of elements in the vector. An assertion to check this would fail in the case where the max value of T1 is larger than the vector size (e.g., for int64_t).